addon/components/frost-table-header.js

```diff
@@ -41,12 +46,115 @@ export default Component.extend({
 
   // == Computed Properties ===================================================
 
+  @readOnly
+  @computed('css')
+  _categoryRowClass (css) {
+    return `${css}-categories`
+  },
+
+  @readOnly
+  @computed('css')
+  _columnRowClass (css) {
+    return `${css}-columns`
+  },
+
+  @readOnly
+  @computed('columns')
+  _hasCategories (columns) {
+    let hasCategories = false
+    columns.forEach((column) => {
+      if (column.category) {
+        hasCategories = true
+        return
+      }
+    })
+    return hasCategories
+  },
+
+  @readOnly
+  @computed('columns')
+  _categoryColumns (columns) {
+    let categoryColumns = A([])
+    let index = 0
+    columns.forEach((column) => {
+      let lastCategoryColumn = categoryColumns.get('lastObject')
+      const categoryLabel = column.category ? column.category : ''
+      if (lastCategoryColumn && lastCategoryColumn.label === categoryLabel) {
+        lastCategoryColumn.span += 1
+      } else {
+        categoryColumns.addObject({
+          index: index++,
+          label: categoryLabel,
+          span: 1,
+          className: column.categoryClassName,
+          renderer: column.headerCategoryRenderer
+        })
+      }
+    })
+    return categoryColumns
+  },
+
   // == Functions =============================================================
 
+  setupRows () {
+    const rowTag = this.get('rowTagName')
+    const selectable = this.get('isSelectable')
+    const categoryColumns = this.get('_categoryColumns')
+    const categoryRowClass = this.get('_categoryRowClass')
+    const columnRowClass = this.get('_columnRowClass')
+
+    // Wrap category and regular header columns into separate rows
+    const lastCategoryIndex = categoryColumns.length + (selectable ? 1 : 0)
+    this.$('.frost-table-header-cell').slice(0, lastCategoryIndex)
+      .wrapAll(`<${rowTag} class='${categoryRowClass} frost-table-row frost-table-header-row'></${rowTag}>`)
+    this.$('.frost-table-header-cell').slice(lastCategoryIndex)
+      .wrapAll(`<${rowTag} class='${columnRowClass} frost-table-row frost-table-header-row'></${rowTag}>`)
+  },
+
+  alignCategories () {
+    const cellTag = this.get('cellTagName')
+    const selectable = this.get('isSelectable')
+    const categoryColumns = this.get('_categoryColumns')
+    const categoryRowClass = this.get('_categoryRowClass')
+    const columnRowClass = this.get('_columnRowClass')
+    const categorySelector = `.${categoryRowClass} .frost-table-cell`
+    const columnSelector = `.${columnRowClass} .frost-table-cell`
+
+    if (cellTag === 'th' || cellTag === 'td') {
+      // Make use of colspan property
+      categoryColumns.forEach((category, index) => {
+        if (selectable) {
```

Comment:
Again the selection checkbox column is not part of the internal column array, so accounting for that column :)